### PR TITLE
Refactor System module for JBallerina

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
Refactor native implementation and unit tests of System module for JBallerina.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/system/src/main/java/org/ballerinalang/stdlib/system/nativeimpl/ReadDir.java [hunk 1]
```diff
@@ -52,35 +56,33 @@
...
         }
-        BMap[] results = null;
+        ObjectValue[] results;
         try (Stream<Path> walk = Files.walk(inputFile.toPath())) {
             results = walk.map(x -> {
                 try {
-                    return getFileInfo(context, x.toFile());
+                    return SystemUtils.getFileInfo(x.toFile());
                 } catch (IOException e) {
                     throw new BallerinaException("Error while accessing file info", e);
```
Comment (chamil321) [inline stdlib/system/src/main/java/org/ballerinalang/stdlib/system/nativeimpl/ReadDir.java]: Shall we use the jvm BallerinaException here?

File: stdlib/system/src/main/java/org/ballerinalang/stdlib/system/utils/SystemUtils.java [hunk 2]
```diff
@@ -87,50 +67,40 @@ public static BError getBallerinaError(String reason, Throwable error) {
...
-        fileInfo.put(FILE_INFO_MODIFIED_TIME_FIELD, lastModifiedInstance);
-        fileInfo.put(FILE_INFO_IS_DIR_FIELD, new BBoolean(inputFile.isDirectory()));
-
-        return fileInfo;
+        ObjectValue fileInfoRecord = BallerinaValues.createObjectValue(SYSTEM_PACKAGE_PATH, FILE_INFO_TYPE);
+        fileInfoRecord.set(FILE_INFO_NAME_FIELD, inputFile.getName());
+        fileInfoRecord.set(FILE_INFO_SIZE_FIELD, inputFile.length());
+        fileInfoRecord.set(FILE_INFO_MODIFIED_TIME_FIELD, lastModifiedInstance);
+        fileInfoRecord.set(FILE_INFO_DIR_FIELD, inputFile.isDirectory());
+        return fileInfoRecord;
```
Comment (chamil321) [inline stdlib/system/src/main/java/org/ballerinalang/stdlib/system/utils/SystemUtils.java]:
```suggestion
        return BallerinaValues.createObjectValue(SYSTEM_PACKAGE_PATH, FILE_INFO_TYPE, inputFile.getName(), inputFile.length(), lastModifiedInstance, inputFile.isDirectory());
```

File: stdlib/system/src/main/ballerina/system/file_info.bal [hunk 3]
```diff
@@ -19,43 +19,44 @@ import ballerina/time;
...
+    boolean dir;
+
+    public function __init(string name, int size, time:Time modifiedTime, boolean dir) {
+        self.name = name;
+        self.size = size;
+        self.modifiedTime = modifiedTime;
+        self.dir = dir;
+    }
+
+    # Returns the filename.
```
Comment (anupama-pathirage) [inline stdlib/system/src/main/ballerina/system/file_info.bal]:
```suggestion
    # Returns the file name.
```

File: stdlib/system/src/main/ballerina/system/file_info.bal [hunk 4]
```diff
@@ -19,43 +19,44 @@ import ballerina/time;
...
+    #
+    # + return - the filename
```
Comment (anupama-pathirage) [inline stdlib/system/src/main/ballerina/system/file_info.bal]:
```suggestion
    # + return - the file name
```

File: stdlib/system/src/main/ballerina/system/file_info.bal [hunk 5]
```diff
@@ -19,43 +19,44 @@ import ballerina/time;
...
+    public function getName() returns string {
+        return self.name;
+    }
+
+    # Returns file size.
```
Comment (anupama-pathirage) [inline stdlib/system/src/main/ballerina/system/file_info.bal]:
```suggestion
    # Returns the file size.
```

**Pull request merged**